wcfsetup/install/files/ts/WoltLabSuite/Core/User.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Provides data of the active user.
+ *
+ * @author  Matthias Schmidt
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  User (alias)
+ * @module  WoltLabSuite/Core/User
+ */
+
+let _link: string | undefined;
+
+export = {
+  /**
+   * Returns the link to the active user's profile or an empty string
+   * if the active user is a guest.
+   */
+  getLink: (): string => _link || '',
+
+  /**
+   * Initializes the user object.
+   */
+  init: (userId: number, username: string, link: string): void => {
+    if (_link !== undefined) {
+      throw new Error('User has already been initialized.');
+    }
+
+    // define non-writeable properties for userId and username
+    Object.defineProperty(this, 'userId', {
+      value: userId,
+      writable: false,
+    });
```

Comment:
Could this be:

```
get userId() {
  return userId;
}
```

plus a standalone variable similar to `_link`?

Reply:
Good catch, I forgot that `get property()` is ES2015 and not TS' syntactic sugar.